lib/compat/wordpress-6.1/template-parts-screen.php

```diff
@@ -0,0 +1,206 @@
+		'supportsLayout'            => WP_Theme_JSON_Resolver::theme_has_support(),
+		'supportsTemplatePartsMode' => current_theme_supports( 'block-template-parts' ),
+		'__unstableHomeTemplate'    => gutenberg_resolve_home_template(),
+	);
+
+	/**
+	 * We don't need home template resolution when block template parts are supported.
+	 * Set the value to true to satisfy the editor initialization guard clause.
+	 */
+	if ( $custom_settings['supportsTemplatePartsMode'] ) {
+		$custom_settings['__unstableHomeTemplate'] = true;
+	}
+
+	// Add additional back-compat patterns registered by `current_screen` et al.
+	$custom_settings['__experimentalAdditionalBlockPatterns']          = WP_Block_Patterns_Registry::get_instance()->get_all_registered( true );
+	$custom_settings['__experimentalAdditionalBlockPatternCategories'] = WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered( true );
+
+	$editor_settings = get_block_editor_settings( $custom_settings, $block_editor_context );
+
+	if ( isset( $_GET['postType'] ) && ! isset( $_GET['postId'] ) ) {
+		$post_type = get_post_type_object( $_GET['postType'] );
+		if ( ! $post_type ) {
+			wp_die( __( 'Invalid post type.', 'gutenberg' ) );
+		}
+	}
+
+	$active_global_styles_id = WP_Theme_JSON_Resolver::get_user_global_styles_post_id();
+	$active_theme            = wp_get_theme()->get_stylesheet();
+	$preload_paths           = array(
+		array( '/wp/v2/media', 'OPTIONS' ),
+		'/wp/v2/types?context=view',
+		'/wp/v2/types/wp_template?context=edit',
+		'/wp/v2/types/wp_template-part?context=edit',
+		'/wp/v2/templates?context=edit&per_page=-1',
+		'/wp/v2/template-parts?context=edit&per_page=-1',
+		'/wp/v2/themes?context=edit&status=active',
+		'/wp/v2/global-styles/' . $active_global_styles_id . '?context=edit',
+		'/wp/v2/global-styles/' . $active_global_styles_id,
+		'/wp/v2/global-styles/themes/' . $active_theme,
+	);
+
+	block_editor_rest_api_preload( $preload_paths, $block_editor_context );
+
+	wp_add_inline_script(
+		'wp-edit-site',
+		sprintf(
+			'wp.domReady( function() {
+				wp.editSite.initializeEditor( "site-editor", %s );
+			} );',
+			wp_json_encode( $editor_settings )
+		)
+	);
+
+	// Preload server-registered block schemas.
+	wp_add_inline_script(
+		'wp-blocks',
+		'wp.blocks.unstable__bootstrapServerSideBlockDefinitions(' . wp_json_encode( get_block_editor_server_block_settings() ) . ');'
+	);
+
+	wp_add_inline_script(
+		'wp-blocks',
+		sprintf( 'wp.blocks.setCategories( %s );', wp_json_encode( get_block_categories( $block_editor_context ) ) ),
+		'after'
+	);
+
+	wp_enqueue_script( 'wp-edit-site' );
+	wp_enqueue_script( 'wp-format-library' );
+	wp_enqueue_style( 'wp-edit-site' );
+	wp_enqueue_style( 'wp-format-library' );
+	wp_enqueue_media();
+
+	if (
+		current_theme_supports( 'wp-block-styles' ) ||
+		( ! is_array( $editor_styles ) || count( $editor_styles ) === 0 )
+	) {
+		wp_enqueue_style( 'wp-block-library-theme' );
+	}
```

Comment:
@Mamaduka I think we may be missing the `enqueue_block_editor_assets` action here. See https://github.com/WordPress/wordpress-develop/blob/52f51f5cdff49e04805233d56560abbbe1813e35/src/wp-admin/site-editor.php#L141

```suggestion
	}
	
	/** This action is documented in wp-admin/edit-form-blocks.php */
	do_action( 'enqueue_block_editor_assets' );
```

Reply:
Nice catch, @fabiankaegy!